### PR TITLE
Don't run tests one-too-many times and remove concurrency cap

### PR DESCRIPTION
**Pull request by elprans**

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -69,10 +70,6 @@ jobs:
         with:
           server-version: ${{ matrix.edgedb-version }}
 
-      - uses: actions-rs/cargo@v1
-        with:
-          command: test
```
Comment (tailhook) [inline .github/workflows/tests.yml:74]: Not sure about this one. Running with glibc is useful. We only need musl build for docker tests. We can run them in parallel or exclude non-docker tests from musl run, though.

Comment (elprans) [reply .github/workflows/tests.yml:74]: But we already run glibc in a sibling job, how is this one different from that?

Comment (tailhook) [reply .github/workflows/tests.yml:74]: Which one? I don't see any.

Comment (elprans) [reply .github/workflows/tests.yml:74]: This?

https://github.com/edgedb/edgedb-cli/blob/496fb2f5b7e55636f130655f42fa2841cbdabe87/.github/workflows/tests.yml#L19-L44

**Review by tailhook — APPROVED** [review]

File: .github/workflows/tests.yml (hunk 1 above)
Comment (tailhook) [reply .github/workflows/tests.yml:74]: Ah, yes sorry. I've thought that should exclude ubuntu, but this way is fine.

**Pull request merged**